src/components/Autocomplete.tsx

```diff
@@ -0,0 +1,112 @@
+import classNames from 'classnames';
+import { Person } from '../types/Person';
+import { useMemo, useState } from 'react';
+import debounce from 'lodash.debounce';
+import './Autocomplete.scss';
+
+type Props = {
+  people: Person[];
+  onSelected: (person: Person | null) => void;
+  delay?: number;
+};
+
+export const Autocomplete: React.FC<Props> = ({
+  people,
+  onSelected = () => {},
+  delay = 300,
+}) => {
+  const [query, setQuery] = useState('');
+  const [focusOnInput, setFocusOnInput] = useState(false);
+  const [appliedQuery, setAppliedQuery] = useState('');
+
+  const applyQuery = debounce((value: string) => {
+    setAppliedQuery(value.trim());
+  }, delay);
+
+  const onInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const value = event.target.value;
+
+    if (value.trim() !== '') {
+      setQuery(value);
+      applyQuery(value);
+    } else {
+      setQuery('');
+      setAppliedQuery('');
+    }
+
+    setFocusOnInput(true);
+    onSelected(null);
+  };
+
+  const onSelectPerson = (person: Person) => {
+    onSelected(person);
+    setFocusOnInput(false);
+    setQuery(person.name);
+  };
+
+  const filteredPeople = useMemo(() => {
+    return people.filter(person =>
+      person.name.toLowerCase().includes(appliedQuery.toLowerCase()),
+    );
+  }, [appliedQuery, people]);
+
+  return (
+    <div
+      className={classNames('dropdown', {
+        'is-active': focusOnInput,
+      })}
+    >
+      <div className="dropdown-trigger">
+        <input
+          type="text"
+          placeholder="Enter a part of the name"
+          className="input"
+          value={query}
+          data-cy="search-input"
+          onFocus={() => setFocusOnInput(true)}
+          onBlur={() => setFocusOnInput(false)}
+          onChange={onInputChange}
+        />
+      </div>
+
+      {focusOnInput && (
+        <div className="dropdown-menu" role="menu" data-cy="suggestions-list">
+          <div className="dropdown-content">
+            {filteredPeople.length > 0 ? (
+              filteredPeople.map(person => (
+                <div
+                  key={person.slug}
```

Comment:
Ensure that the `slug` property used as a key is unique and consistent for each `Person` object. This is important to avoid potential rendering issues in React. If `slug` is not guaranteed to be unique, consider using another unique identifier.